oat-sa/tao-core-ui-fe
# Feature/tao 10457/implement class filter

**related to:** https://oat-sa.atlassian.net/browse/TAO-10457
**Required by**: https://github.com/oat-sa/tao-core/pull/2668

**description:** `searchModal` component now supports a class filter input provided by `resourceSelector` component. Feature is only enabled on certain TAO `contexts`, so if no `rootClassUri` is provided class filter is disabled. Class filter value is stored on a new `criterias` property on `searchStore` where all future criterias will be saved.

**How to test**
To be able to test this PR locally you need to setup your TAO instance to manage the new Advanced search feature. To do so you need to:

On your root `composer.json` set the following dependencies:
```
    "oat-sa/tao-core": "dev-feature/TAO-10457/implement-class-filter",
    "oat-sa/generis": "dev-feature/TAO-10203-advanced-search as 12.33.2",
    "oat-sa/extension-tao-dac-simple": "dev-feature/TAO-10203-advanced-search",
    "oat-sa/lib-tao-elasticsearch": "dev-feature/TAO-10203-advanced-search as 1.1.1"
```
run `composer install`
if you work with taoDockerize run `taoDockerize install `(database will be reset)
If you do not user taoDockerize run `php tao/scripts/taoUpdate.php`
Now `searchModal` component should be used on search trigger instead of previous search component. Notice that advanced search syntax is not enabled if you do not set up an elastic search server, which is not required to test this feature.

Please let me know if you need more details about how to test the feature.

![class-filter](https://user-images.githubusercontent.com/14041944/92386677-8159b000-f114-11ea-86ea-f739cdb59bea.gif)
